azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/strategy/CreateResourcesThenCreateNodes.java

```diff
@@ -50,7 +50,7 @@
 import org.jclouds.azurecompute.arm.domain.Subnet.SubnetProperties;
 import org.jclouds.azurecompute.arm.domain.VirtualNetwork.AddressSpace;
 import org.jclouds.azurecompute.arm.domain.VirtualNetwork.VirtualNetworkProperties;
-import org.jclouds.azurecompute.arm.util.Passwords;
+import org.jclouds.util.Passwords;
 import org.jclouds.compute.config.CustomizationResponse;
 import org.jclouds.compute.domain.NodeMetadata;
 import org.jclouds.compute.domain.Template;
```

Comment:
Short answer: yes
Couldn't find authoritative docs on what the exact requirements for the password are. The generator is pretty conservative with the characters used and testing the output with Azure works fine.
I'd do one more change with the password generator, but in a separate PR - use a fixed max length for the password. Currently it will pick a length between 8 and 50 characters. Don't think it adds any security. I'd go with the MAX always for consistency. It will also bring out any problems with character length support in the providers (say if tests have been passing, but the password being generated is not in the extremes).